Backend/mentii/user_ctrl.py

```diff
@@ -172,16 +174,81 @@ def getUserByEmail(email, dbInstance):
 
   table = dbUtils.getTable('users', dbInstance)
   if table is None:
-    MentiiLogging.getLogger().error("Unable to get table users in getUserByEmail")
+    MentiiLogging.getLogger().error('Unable to get table users in getUserByEmail')
     return None
 
-  key = {"Key" : {"email": email}}
+  key = {'Key' : {'email': email}}
   result = dbUtils.getItem(key, table)
   if result is None:
-    MentiiLogging.getLogger().error("Unable to get the user with email: " + email + " in getUserByEmail ")
+    MentiiLogging.getLogger().error('Unable to get the user with email: ' + email + ' in getUserByEmail ')
     return None
 
   if 'Item' in result.keys():
     user = result['Item']
 
   return user
+
+def changeUserRole(jsonData, dbInstance, adminRole=None):
+  response = ControllerResponse()
+
+  #g will be not be avliable durring testing
+  #and adminRole will need to be passed to the function
+  if g :
+    adminRole = g.authenticatedUser['userRole']
+  #adminRole is confirmed here incase changeUserRole is called from somewhere
+  #other than app.py changeUserRole()
+  if adminRole != 'admin':
+    response.addError('Role Error', 'Only admins can change user roles')
+  elif 'email' not in jsonData.keys() or 'userRole' not in jsonData.keys():
+    response.addError('Key Missing Error', 'Email or role missing from json data')
+  else:
+    email = jsonData['email']
+    userRole = jsonData['userRole']
+
+    userTable = dbUtils.getTable('users', dbInstance)
+    if userTable is None:
+      MentiiLogging.getLogger().error('Unable to get table "users" in changeUserRole')
+      response.addError('No Access to Data', 'Unable to get data from database')
+    else:
+      if userRole != 'student' and userRole != 'teacher' and userRole != 'admin':
+        MentiiLogging.getLogger().error('Invalid role: ' + userRole + ' specified. Unable to change user role')
+        response.addError('Invalid Role Type', 'Invaid role specified')
+      else:
+
+        data = {
+            'Key': {'email': email},
+            'UpdateExpression': 'SET userRole = :ur',
+            'ExpressionAttributeValues': { ':ur': userRole },
+            'ReturnValues' : 'UPDATED_NEW'
+        }
+
+        result = dbUtils.updateItem(data, userTable)
+
+        if result is None:
+          MentiiLogging.getLogger().error('Unable to update the user with email: ' + email + ' in changeUserRole')
+          response.addError('Result Update Error', 'Could not update the user role in database')
+        else:
+          response.addToPayload('Result:', result)
+          response.addToPayload('success', 'true')
+
+  return response
+
+def getRole(userEmail, dynamoDBInstance):
+  '''
+  Returns the role of the user whose email is pased. If we are unable to get
+  this information from the DB the role 'student' is returned
```

Comment:
This originally was done to prevent another error check at the person calling this. When I wrote the function I didn't think this would be our big security check and seeing as student is the lowest priority user it wouldn't be a bad classification. While it's still the case that this isn't the big security check we can make that change.